tools/deps.nim

```diff
@@ -1,13 +1,22 @@
-import os, uri, strformat
+import os, uri, strformat, osproc
 
 proc exec(cmd: string) =
   echo "deps.cmd: " & cmd
   let status = execShellCmd(cmd)
   doAssert status == 0, cmd
 
+proc execEx(cmd: string): tuple[output: TaintedString, exitCode: int] =
+  echo "deps.cmd: " & cmd
+  execCmdEx(cmd, {poStdErrToStdOut, poUsePath, poEvalCommand})
+
+proc isGitRepo(dir: string): bool =
+  let (output, status) = execEx fmt"git -C {quoteShell(dir)} rev-parse --show-cdup"
```

Comment:
Please document what this git gibberish means for us mortals who seek to know as little about git as possible.